### PR TITLE
Feat/node lifecycle unit test

### PR DESCRIPTION
**Pull request by DarkKnight3074**
This PR adds comprehensive test coverage for node lifecycle management and initialization, implements robust error handling for both leader and regular node initialization, and adds functionality for leader nodes to handle operator deactivation events.

  Changes

  Test Coverage Enhancements

  - Leader Node Tests (+2428 lines): Added extensive test cases for leader node initialization scenarios including successful initialization, database failures, contract interaction errors, libp2p connection handling, and various failure recovery scenarios
  - Regular Node Tests (+1698 lines): Implemented comprehensive test coverage for regular node lifecycle including initialization, error scenarios, connection management, and graceful failure handling
  - Handler Tests (+660 lines): Expanded test coverage for leader node handlers, focusing on libp2p connection scenarios, event handling, and edge cases
  - Accept Commit Tests (+288 lines): Added test cases for commit acceptance flow in leader nodes

  Features

  - Leader Node Error Handling: Implemented robust error handling in leader node initialization process, ensuring graceful failure and proper resource cleanup (nodes/leader/leader_node.go:32, cmd/nodes/main.go:10)
  - Regular Node Error Handling: Added comprehensive error handling in regular node initialization process (nodes/regular/regular_node.go:30)
  - Operator Deactivation Handling: Leader nodes now properly disconnect from regular nodes when receiving deactivation events from the contract (nodes/leader/accept_commit.go:35)

  Bug Fixes & Improvements

  - Enhanced libp2p connection test coverage to ensure stable peer-to-peer communication
  - Improved error propagation and handling across both leader and regular node initialization flows
  - Updated handler logic for better connection state management
  - Better resource cleanup on initialization failures

  Files Changed

  - nodes/leader/leader_node.go - Error handling implementation
  - nodes/leader/leader_node_test.go - Extensive initialization test cases
  - nodes/leader/handler.go - Handler improvements
  - nodes/leader/handler_test.go - Enhanced handler testing
  - nodes/leader/accept_commit.go - Deactivation event handling
  - nodes/leader/accept_commit_test.go - Commit flow validation
  - nodes/regular/regular_node.go - Improved error handling
  - nodes/regular/regular_node_test.go - Complete lifecycle test coverage
  - cmd/nodes/main.go - Updated initialization flow

  Test Plan

  - All new test cases pass locally
  - Leader node initialization handles various failure scenarios gracefully
  - Regular node initialization with error handling works as expected
  - Both nodes properly clean up resources on initialization failures
  - Deactivation events properly trigger peer disconnection
  - Libp2p connection stability verified through extensive test coverage

### CODE REVIEW

File: nodes/leader/accept_commit.go [hunk 1]
```diff
@@ -314,8 +315,40 @@ func (n *LeaderNode) processDeactivated(ctx context.Context, operator common.Add
...
+		if err == nil {
+			for _, nodeInfo := range nodeInfos {
+				if nodeInfo.EOAAddress == operator.Hex() {
+					peerIDStr = nodeInfo.PeerID
+					break
+				}
+			}
+		}
+
+		if peerIDStr != "" && n.p2pClient != nil {
```
Comment (ghost) [inline nodes/leader/accept_commit.go]: could you use early return?

**Review by ghost — COMMENTED** [review]
One comment

**Review by Mehd1b — APPROVED** [review]

File: nodes/leader/accept_commit.go (hunk 1 above)
Comment (DarkKnight3074) [reply nodes/leader/accept_commit.go]: Could you elaborate?

File: nodes/leader/accept_commit.go [hunk 2]
```diff
@@ -314,8 +315,40 @@ func (n *LeaderNode) processDeactivated(ctx context.Context, operator common.Add
...
 	if n.nodeInfoRepository != nil {
+		nodeInfos, err := n.nodeInfoRepository.GetNodeInfos(ctx)
+		if err == nil {
+			for _, nodeInfo := range nodeInfos {
+				if nodeInfo.EOAAddress == operator.Hex() {
+					peerIDStr = nodeInfo.PeerID
+					break
+				}
+			}
+		}
```
Comment (ghost) [inline nodes/leader/accept_commit.go]:
```suggestion
        nodeInfos, err := n.nodeInfoRepository.GetNodeInfos(ctx)
        if err != nil {
		            logger.Errorf("Failed to get nodes info, err: %v", err)
		            return err
        }
        for _, nodeInfo := range nodeInfos {
		        if nodeInfo.EOAAddress == operator.Hex() {
			        peerIDStr = nodeInfo.PeerID
			        break
		        }
	        }
	}
	
	if peerIDStr == "" {
		  // Handle if the peer id isnt stored on the db
	}
	
	hostInstance := n.p2pClient.GetHostInstance()
	if hostInstance == nil {
		// Handle
	}
	....
```

sth like that, and i think we don't need to check n.p2pClient, n.nodeInfoRepository with nil, they should be non-empty in the constructor

File: nodes/leader/accept_commit.go (hunk 1 above)
Comment (ghost) [reply nodes/leader/accept_commit.go]: I commented below

File: nodes/leader/accept_commit.go (hunk 2 above)
Comment (DarkKnight3074) [reply nodes/leader/accept_commit.go]: I have removed it.
